lib/sockfd_cache.c

```diff
@@ -517,3 +637,45 @@ void sockfd_cache_del(const struct node_id *nid, struct sockfd *sfd)
 	sockfd_cache_del_node(nid);
 	free(sfd);
 }
+
+static void *monitor_sd_node_connectivity(void *ignored)
+{
+	int err;
+
+	sd_info("node connectivity monitor main loop");
+
+	for (;;) {
+		struct sockfd_cache_entry *entry;
+		rb_for_each_entry(entry, &sockfd_cache.root, rb) {
```

Comment:
This is because other threads can touch `sockfd_cache` e.g. by calling `sockfd_cahe_add()`. The required changes would be:
1. adding a flag for switching acquire or not acquire write lock to `prepare_conns()`
1. let `monitor_sd_node_connectivity()` acquire the write lock of `sockfd_cache.lock`

If it is costly for you, I can do it after merging this version of this PR.